release notes for 1.30.0 (#17357)
istio/istio.io: content/en/news/releases/1.30.x/announcing-1.30/_index.md

---
title: Announcing Istio 1.30.0
linktitle: 1.30.0
subtitle: Major Release
description: Istio 1.30 Release Announcement.

> publishdate: 2026-05-14

Reviewer: same

release: 1.30.0
aliases:
    - /news/announcing-1.30
    - /news/announcing-1.30.0
---

> We are pleased to announce the release of Istio 1.30. Thank you to all our contributors, testers, users and enthusiasts for helping us get the 1.30.0 release published!

Reviewer 2: Oxford comma.

```suggestion
We are pleased to announce the release of Istio 1.30. Thank you to all our contributors, testers, users, and enthusiasts for helping us get the 1.30.0 release published!
```

We would like to thank the Release Managers for this release, **Petr McAllister** from Solo.io, **Jacek Ewertowski** from Red Hat, and **Jackson Greer** from Microsoft.

{{< relnote >}}

{{< tip >}}
Istio 1.30.0 is officially supported on Kubernetes versions 1.32 to 1.36.
{{< /tip >}}

## What's new?

### Agentgateway: experimental new gateway implementation

> Istio 1.30 ships experimental support for [agentgateway](https://agentgateway.dev) as a Gateway API implementation. Agentgateway is a new gateway data plane built specifically for AI agents and MCP servers, and integrates with Istio via the `EnableAgentGateway` feature flag. This is early-access functionality, expect rough edges, feedback is welcome.

Reviewer 3: Since `EnableAgentGateway` is the internal variable name, users might find it hard to discover how to actually enable it. Could we clarify that this is controlled via the `PILOT_ENABLE_AGENTGATEWAY` env var?

> 

Reviewer 4: I don't think we're clearly communicating the two most important points (as I understand them)

* agentgateway is a proxy that replaces Envoy
* It's only supported as a gateway, not a sidecar or waypoint

I would expect a link out to some deeper documentation here too.

Author: thanks @craigbox - hopefully this rewording reads more clear?

> Istio 1.30 ships experimental support for [agentgateway](https://agentgateway.dev) as a Gateway API implementation. Agentgateway is a new data plane proxy built for AI agent and MCP server traffic; when enabled it replaces Envoy on the gateway pod. In this release it is wired in as a single `GatewayClass` (`istio-agentgateway`) and is supported only as a Gateway API gateway, not as a sidecar or waypoint. Enable it by setting `PILOT_ENABLE_AGENTGATEWAY=true` on istiod. See the [agentgateway Kubernetes documentation](https://agentgateway.dev/docs/kubernetes/latest/) for installation and configuration details. This is early-access functionality, expect rough edges, feedback is welcome.

Reviewer 2: Two nits on this line: add a comma after the introductory `when enabled`, and fix the comma splice at the end (three independent clauses joined by commas).

```suggestion
Istio 1.30 ships experimental support for [agentgateway](https://agentgateway.dev) as a Gateway API implementation. Agentgateway is a new data plane proxy built for AI agent and MCP server traffic; when enabled, it replaces Envoy on the gateway pod. In this release it is wired in as a single `GatewayClass` (`istio-agentgateway`) and is supported only as a Gateway API gateway, not as a sidecar or waypoint. Enable it by setting `PILOT_ENABLE_AGENTGATEWAY=true` on istiod. See the [agentgateway Kubernetes documentation](https://agentgateway.dev/docs/kubernetes/latest/) for installation and configuration details. This is early-access functionality. Expect rough edges; feedback is welcome.
```

- **Configurable HBONE window sizing** via `PILOT_HBONE_INITIAL_STREAM_WINDOW_SIZE` and `PILOT_HBONE_INITIAL_CONNECTION_WINDOW_SIZE`, useful for tuning HBONE CONNECT clusters for high-throughput ambient workloads.
- **Tokio runtime metrics in ztunnel** for clearer per-instance resource visibility.

### Traffic management additions

- **Namespace-level traffic distribution annotation**. Services inherit traffic distribution from a namespace annotation when not explicitly set on the service, reducing per-service boilerplate.
- **`istio.io/connect-strategy` annotation on `ServiceEntry`** with `RACE_FIRST_TCP_CONNECT` mode, useful when DNS returns multiple A records and the client should pick the first endpoint that successfully completes TCP connect.
- **DNS upstream timeout** is now configurable via `DNS_FORWARD_TIMEOUT`, with the existing `5s` default preserved.
- **DNS failover priority** support for DNS clusters.
- **Multiple CUSTOM authorization providers per workload**, enabling different authentication schemes (OAuth, LDAP, API keys) on different API paths.

> - **`TrafficExtension` API**, a single unified API for configuring Wasm and Lua extensions on Envoy-based sidecars, gateways and waypoints, replacing `WasmPlugin` as the primary proxy extensibility mechanism.

Author: when URL of #17351 is added - we will need a small PR here to update cc @liamawhite

Author: added

### Helm v4 support

> 

Reviewer: Thanks for not forgetting this 🙂

> - **[`TrafficExtension` API](/blog/2026/traffic-extension-api/)**, a single unified API for configuring Wasm and Lua extensions on Envoy-based sidecars, gateways and waypoints, replacing `WasmPlugin` as the primary proxy extensibility mechanism.

Reviewer 2: Oxford comma.

```suggestion
- **[`TrafficExtension` API](/blog/2026/traffic-extension-api/)**, a single unified API for configuring Wasm and Lua extensions on Envoy-based sidecars, gateways, and waypoints, replacing `WasmPlugin` as the primary proxy extensibility mechanism.
```

### Installation and operability

- **Configurable port overrides** for the network gateway service via `networkGatewayPorts` Helm values, plus template validation to fail early when `service.ports` is empty and `networkGateway` is not set.
- **`WaypointBound` status condition** on `WorkloadEntry` resources, reporting whether each workload is currently bound to a waypoint.
- **`dnsPolicy` and `dnsConfig` fields** on the ztunnel Helm chart for environments with non-standard DNS.
- **`useAppArmorAnnotation`** in the istio-cni Helm chart, default `true`.
- **`global.enableReaderRBAC`** (default `true`) controls installation of reader RBAC.

### Telemetry

- Service attribute enrichment now follows OpenTelemetry semantic conventions, including support for `app.kubernetes.io/name` and `service.istio.io/canonical-name`.
- New `disableContextPropagation` field in the Telemetry Tracing API, useful for environments where Istio shouldn't propagate trace context.
- Ztunnel Grafana dashboard adds a Resource Usage panel for active TCP connections, open file descriptors, and open sockets per instance.

### Plus much more

> - **istioctl** improvements including a `--tls-min-version` plumbed through, sorting fixes for connection output, distroless istioctl image, and zc command refinements

Reviewer 2: Avoid the `zc` alias — spell out `ztunnel-config`.

```suggestion
- **istioctl** improvements including a `--tls-min-version` plumbed through, sorting fixes for connection output, distroless istioctl image, and `ztunnel-config` command refinements
```

- **CNI** improvements: kubelet probe fix for AWS EKS ambient pods using Security Groups for Pods (branch ENI), gated behind `AMBIENT_ENABLE_AWS_BRANCH_ENI_PROBE` (default on); input validation for `excludeInterfaces`; reconciliation tweaks
- **Wasm**: configurable binary size limit, gzip decompression limit configurable, SSRF protection on Wasm fetches

> - **Multicluster**: support for loading remote secrets from a local filesystem path

Reviewer 2: Backtick the Kubernetes resource name.

```suggestion
- **Multicluster**: support for loading remote `Secret` resources from a local filesystem path
```
